sdk/core/azure-core/azure/core/pipeline/policies/_utils.py

```diff
@@ -25,21 +25,24 @@
 # --------------------------------------------------------------------------
 import datetime
 import email.utils
+from typing import Optional
+
 from ...utils._utils import _FixedOffset, case_insensitive_dict
 
 
-def _parse_http_date(text):
+def _parse_http_date(text: str) -> datetime.datetime:
     """Parse a HTTP date format into datetime."""
     parsed_date = email.utils.parsedate_tz(text)
-    return datetime.datetime(*parsed_date[:6], tzinfo=_FixedOffset(parsed_date[9] / 60))
+    return datetime.datetime(*parsed_date[:6], tzinfo=_FixedOffset(parsed_date[9] / 60))  # type: ignore
 
 
-def parse_retry_after(retry_after):
+def parse_retry_after(retry_after: str) -> float:
     """Helper to parse Retry-After and get value in seconds.
 
     :param str retry_after: Retry-After header
-    :rtype: int
+    :rtype: float
```

Comment:
I would suggest that we return `int` rather than `float`. The only time we returned `float` in the past was if the `Retry-After` header was in a HTTP date-time format (which we don't do in Azure), and rounding that up should be fine.

Reply:
We do support the HTTP date parsing in that function, I assume we added it for a reason, or a service?

Reply:
Then I would `int() + 1` the returned value for a parsed HTTP date.

Reply:
https://github.com/Azure/azure-sdk-for-python/issues/31004